transport/tchannel/header.go

```diff
@@ -118,17 +102,24 @@ var emptyMap = map[string]string{}
 // InboundCallResponse.
 //
 // If the format is JSON, the headers are JSON encoded.
-func writeHeaders(format tchannel.Format, headers transport.Headers, getWriter func() (tchannel.ArgWriter, error)) error {
+func writeHeaders(format tchannel.Format, headers map[string]string, tracingBaggage map[string]string, getWriter func() (tchannel.ArgWriter, error)) error {
+	merged := make(map[string]string, len(headers)+len(tracingBaggage))
```

Comment:
can you move this into a function, that way we can optimize the case where `headers` or `tracingBaggage` is empty (which is the case for response headers).
```
func mergeHeaders(m1, m2 map[string]string) map[string]string {
  if len(m1) == 0 {
    return m2
  }
  if len(m2) == 0 {
    return m1
  }
  // merge and return
}
```